report_qweb_table_pagebreak/models/ir_qweb.py

```diff
@@ -0,0 +1,104 @@
+from lxml import html
+
+from odoo import api, models
+
+from odoo.addons.base.models.qweb import QWeb
+
+
+class StyleParser:
+    """Class to manipulate and Serialize/DeSerialize style attribute on
+    html tag::
+
+        style = StyleParser(element.get("style"))
+        # set attribute page-break-inside
+        style.set("page-break-inside", "avoid")
+        # Serialize style to set style value on html element
+        element.set("style", str(style))
+    """
+
+    _styles: dict = {}
+
+    def __init__(self, style_content: str):
+        self._parse(style_content)
+
+    def _parse(self, style_content):
+        self._styles = {}
+        for style in style_content.split(";"):
+            data_style = style.split(":")
+            if len(data_style) != 2:
+                # ignore if not expected string
+                continue
+            key, value = data_style
+            self._styles[key.strip()] = value.strip()
+
+    def __str__(self):
+        return "; ".join([f"{k}: {v}" for k, v in self._styles.items()])
+
+    def set(self, key, value, default=None):
+        if value is None:
+            value = default
+        if value is None:
+            self._styles.pop(key, None)
+        else:
+            self._styles[key] = value
+
+
+class IrQWeb(models.AbstractModel, QWeb):
+    _inherit = "ir.qweb"
+    _description = "Qweb"
+
+    @api.model
+    def _render(self, id_or_xml_id, values=None, **options):
+        result = super()._render(id_or_xml_id, values=values, **options)
+
+        if b"data-page-break=" not in result:
+            return result
+
+        return self._split_tables(result)
+
+    @api.model
+    def _split_tables(self, html_content):
+        html_document = html.document_fromstring(html_content.decode("utf-8"))
+
+        for fragment in html_document:
+            table = None
+            for row in fragment.iterfind(".//tr[@data-page-break]"):
+                # while generating pdf this method is called twice so we prevent
+                # table to be split twice by removing data-page-break once
+                # table has been spited
+                pos = row.attrib.pop("data-page-break")
+                if pos not in ("before", "after", "avoid"):
+                    continue
+                table = next(row.iterancestors("table"))
+                tbody = next(row.iterancestors("tbody"))
+                newtable = html.Element("table", attrib=dict(table.attrib))
+                newtbody = html.Element("tbody", attrib=dict(tbody.attrib))
+                thead = table.find("thead")
+                if thead:
+                    new_thead = html.fromstring(html.tostring(thead))
+                    newtable.append(new_thead)
```

Comment:
I haven't tried this but what if the table fits in one page? Is this repeating the `thead` for each row having `data-page-break`?

Reply:
Yes, for any row that define a valid value for `data-page-break` this will split table repeating the header on each tables. When using `avoid` you can get two tables on the same page, column width can be different in such case. 

ps: you can test it on runboat with this PR https://github.com/OCA/sale-reporting/pull/243 which split table before each sections !